addons/hr_timesheet/security/hr_timesheet_security.xml

```diff
@@ -30,6 +30,25 @@
             <field name="groups_id" eval="[(4,ref('group_timesheet_manager'))]"/>
         </record>
 
+        <record id="timesheet_line_rule_portal_user" model="ir.rule">
+            <field name="name">account.analytic.line.timesheet.portal.user</field>
+            <field name="model_id" ref="analytic.model_account_analytic_line"/>
+            <field name="domain_force">[
+                ('project_id', '!=', False),
+                '|',
+                    ('project_id.message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                    ('task_id.message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id]),
+                ('project_id.allow_project_sharing', '=', True),
+                ('project_id.privacy_visibility', '=', 'portal'),
+                ('project_id.project_sharing_access_ids.user_id', 'in', [user.id]),
+            ]</field>
+            <field name="perm_read" eval="True"/>
+            <field name="perm_write" eval="False"/>
+            <field name="perm_create" eval="False"/>
+            <field name="perm_unlink" eval="False"/>
```

Comment:
you can just leave the default values to True on all perms, that way if someone modifies the ACL to give C/W/U permissions, this rule will still give some protection.